17-lifecycle-lab/src/01-cps.jsx

```diff
@@ -1,18 +1,63 @@
-/**
- * Lab 01: Click per seconds
- *
- * Write a component that counts clicks per second.
- * Component should show a button and some text, and print
- * the click-per-seconds rate.
- */
-
-import React from 'react';
-
-export default createClass({
-  render() {
-    return (<div>
-      <button>Click Fast</button>
-      <p>CPS rate: {0}</p>
-    </div>);
-  }
-});
+/**
+ * Lab 01: Click per seconds
+ *
+ * Write a component that counts clicks per second.
+ * Component should show a button and some text, and print
+ * the click-per-seconds rate.
+ */
+
+import React from 'react';
+
+export default class CPS extends React.Component{
+  constructor(props) {
+      super(props);
+      this.state = {
+        clicks: 0,
+        seconds: 0,
+      };
+  }
+
+  buttonClick = (ev) => {
+      let clicks = this.state.clicks + 1;
+      this.setState({ clicks });
+  };
+
+  message = (ev) => {
+      if (this.state.clicks > this.state.seconds * 2.1) {

```

Comment:
It's better to destructure all needed data from the state at the beginning of the method, so you can easily see what state variables are used.